runtime/ops/encoding.rs

```diff
@@ -0,0 +1,129 @@
+//! Non-web API ops that deal with encoding and decoding of data.
+use deno_core::op;
+use deno_core::Extension;
+
+pub fn init() -> Extension {
+  Extension::builder()
+    .ops(vec![op_escape_html::decl()])
+    .build()
+}
+
+// TODO(@littledivy): Fast path for Latin1 strings
+#[op]
+fn op_escape_html(s: String) -> String {
+  let mut out = Vec::with_capacity(s.len());
```

Comment:
yes - i think would be good to check - it might be just as quick to do so in JS using a regex as the JS HTMLEscape does? this would avoid crossing boundary into Rust altogether if we have no work to do. Currently this code is 25m ops/sec for a 10 character string of zeros (no escaping to do) if we test the regex first versus 5m ops/sec if we don't.
```JavaScript
  if (Deno.env.get('INTERNAL')) {
    HTMLEscape = str => {
      if (!ma.exec(str)) return str
      return Deno.escapeHtml(str)
    }
  }
```

Reply:
i also tested this technique on [v_escape](https://github.com/botika/v_escape) rust library and see same difference if we test the regex in JS first. once again though - the regex technique is **only faster for small strings** - from testing i have done it is quicker to skip the regex check for strings longer than 256 bytes.

Reply:
did some more testing and here are the tentative conclusions
- if SIMD is available, then testing for a non-matching string in JS is faster up to a payload size of 512 bytes
- if SIMD is not available, testing for a non-matching string in JS is *always* faster than calling into Rust - in fact, doing the escaping in JS is always faster in this case.

i'll put some benchmark results up in a bit.

Reply:
Another fast path might be to mutate the string in-place?